app/controllers/spree/admin/products_controller.rb

```diff
@@ -9,36 +9,34 @@ class ProductsController < ResourceController
       include OpenFoodNetwork::SpreeApiKeyLoader
       include OrderCyclesHelper
       include EnterprisesHelper
-
-      create.before :create_before
-      update.before :update_before
+      include ProductFilterHelper
 
       before_action :load_data
       before_action :load_form_data, only: [:index, :new, :create, :edit, :update]
       before_action :load_spree_api_key, only: [:index, :variant_overrides]
       before_action :strip_new_properties, only: [:create, :update]
 
-      respond_override create: { html: {
-        success: lambda {
-          if params[:button] == "add_another"
-            redirect_to new_admin_product_path
-          else
-            redirect_to admin_products_path
-          end
-        },
-        failure: lambda {
-          render :new
-        }
-      } }
-
       def new
         @object.shipping_category = DefaultShippingCategory.find_or_create
         super
```

Comment:
same as the edit action, instead of a call to super we can probably have a single line here using render.

Reply:
I was wondering, do we actually need to use `respond_with(@object)`  ? if not we can just get rid of the super.

Reply:
no we dont need to use it. We can bypass it as long as things continue to work. I am not sure exactly what it does... it's from spree. Maybe it's just something that can be overrided with respond_override? I am not sure, I'd love to get rid of it:
https://github.com/openfoodfoundation/spree/blob/2-1-0-stable/core/lib/spree/core/controller_helpers/respond_with.rb

Reply:
I removed the respond_with where it wasn't needed.